test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -90,7 +90,7 @@ package object test extends CheckVariants {
   /**
    * An `ExecutedSpec` is a spec that has been run to produce test results.
    */
-  type ExecutedSpec[+L, +E, +S] = Spec[Any, Nothing, L, Either[TestFailure[E], TestSuccess[S]]]
+  type ExecutedSpec[+L, +E, +S] = Spec[Any, Nothing, L, (Either[TestFailure[E], TestSuccess[S]], TestAnnotationMap)]
```

Comment:
I like the idea that executing the spec gives you back the annotation map (at each node) always. 👍 

It may be empty in some configurations, that's fine.

By the way, this structure is _beginning_ to get complex enough that we should consider a custom type:

```scala
sealed trait TestResult[+E, +S] {
  def annotations: TestAnnotationMap
}
object TestResult {
  final case class Failed[E](failure: TestFailure[E], annotations: TestAnnotationMap) extends TestResult[E, S]
  final case class Succeeded[A](success: TestSuccess[S], annotations: TestAnnotationMap) extends TestResult[E, S]
}
```

Then we can simplify to:

```scala
type ExecutedSpec[+L, +E, +S] = 
  Spec[Any, Nothing, L, TestResult[E, S]]
```

Reply:
(I think we're already using `TestResult` name for something...)

Reply:
Yes. After getting this merged my next priority is to simplify some of these types.